test/OpenTelemetry.Exporter.OneCollector.Tests/CommonSchemaJsonSerializationStateTests.cs

```diff
@@ -78,6 +78,39 @@ public void AddExtensionAttributeTest()
             json);
     }
 
+    [Fact]
+    public void AddExtensionAttributeWithComplexTypeTest()
+    {
+        using var stream = new MemoryStream();
+        using var writer = new Utf8JsonWriter(stream);
+
+        var state = new CommonSchemaJsonSerializationState("Test", writer);
+
+        state.AddExtensionAttribute(new KeyValuePair<string, object?>(
```

Comment:
Consider having another test case:

```csharp

var metadata = new Dictionary<string, object>
               {
                   ["f"] = new Dictionary<string, object>
                   {
                       ["Id"] = 1234,
                   },
               };
state.AddExtensionAttribute(new KeyValuePair<string, object?>("ext.metadata", metadata);

metadata["x"] = 1; // modify the object after AddExtensionAttribute

// ...

Assert.Equal(
            "{\"ext\":{\"metadata\":{\"f\":{\"Id\":1234}}}}",
            json);
```

Reply:
Some bad news here. The OTel SDK will copy/buffer anything exposed by the TState logged if it implements `IReadOnlyList` or `IEnumerable` of `KeyValuePair<string, object>` but ONLY the top-level things. It won't recurse into the values (`object?`s) of the top-level things and attempt to buffer reference types. I think best to tackle this in the SDK itself?

Reply:
> It won't recurse into the values (`object?`s) of the top-level things and attempt to buffer reference types. I think best to tackle this in the SDK itself?

I actually don't think the SDK should do anything about these at all. Think about how OS kernels are going to handle interrupts.